kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/api/UploadBackendClient.kt

```diff
@@ -94,6 +96,8 @@ class UploadBackendClient(
             when (response.code()) {
                 401 -> throw NotAuthorizedException("access token is not provided or is invalid : ${response.message()}")
                 403 -> throw NotAuthorizedException("access token is not authorized to perform this request")
+                400 -> throw StaleStateException("Could not perform request due stale state: ${response.message()}")
```

Comment:
I'm not sure if 400 is the right code for a stale state, the request is well-formed, the server just can't process it. Perhaps 425 Too Early or even just 409 Conflict may be better candidates.

Reply:
Then I would go for the 409. In this case, request is wellformed though it has an old state in the context on the server.